react/src/components/atoms/forms/InputCurrency/index.js

```diff
@@ -47,148 +47,130 @@ const Currency = (props) => {
-            } else {
-              numberValue = Number(numbro.unformat(stringValue));
-            }
+            const numberValue = stringValue ? Number(numbro.unformat(stringValue)) : 0;
             // default to 0 if defaultValue is NaN
             if (is.number(numberValue) && !is.empty(stringValue)) {
               let newValue = numberValue;
               if (key === 'ArrowDown') {
                 newValue = Number(numbro(numberValue).subtract(props.step).format({ mantissa: countDecimals(props.step) }));
-                if ((!hasProperty(props, 'min') || newValue >= props.min) && (!hasProperty(props, 'max') || (newValue <= props.max))) {
-                  const { showError, errorMsg } = validNumber(newValue, props.min, props.max);
-                  context.updateState({
-                    showError,
-                    errorMsg,
-                    value: toCurrency(newValue, countDecimals(props.step))
-                  }, () => {
-                    if (typeof props.onChange === 'function') {
+                if (greaterThanMin(newValue) && lessThanMax(newValue)) {
+                  const updateError = displayErrorMessage(!is.empty(stringValue) ? newValue : '');
+                  context.updateState({ value: toCurrency(newValue, countDecimals(props.step)), ...updateError }, () => {
+                    if (is.fn(props.onChange)) {
                       props.onChange(newValue, props.id, type, key);
                     }
                   });
                 }
               } else if (key === 'ArrowUp') {
                 newValue = Number(numbro(numberValue).add(props.step).format({ mantissa: countDecimals(props.step) }));
-                if ((!hasProperty(props, 'min') || newValue >= props.min) && (!hasProperty(props, 'max') || (newValue <= props.max))) {
-                  const { showError, errorMsg } = validNumber(newValue, props.min, props.max);
-                  context.updateState({
-                    showError,
-                    errorMsg,
-                    value: toCurrency(newValue, countDecimals(props.step))
-                  }, () => {
-                    if (typeof props.onChange === 'function') {
+                if (greaterThanMin(newValue) && lessThanMax(newValue)) {
+                  const updateError = displayErrorMessage(!is.empty(stringValue) ? newValue : '');
+                  context.updateState({ value: toCurrency(newValue, countDecimals(props.step)), ...updateError }, () => {
+                    if (is.fn(props.onChange)) {
                       props.onChange(newValue, props.id, type, key);
                     }
                   });
                 }
               }
             }
           };
+
           const handleBlur = () => {
             const inputEl = ref.current;
-            if (is.empty(inputEl.value)) {
+            const stringValue = inputEl.value;
+            const isNotNumber = !stringValue || isNaN(Number(numbro.unformat(stringValue)));
```

Comment:
```suggestion
            const isNotNumber = !stringValue || Number.isNaN(Number(numbro.unformat(stringValue)));
```

Reply:
i think this is doing the same thing

Reply:
because stringValue can't be number 0

Reply:
Number.isNaN() is the same as isNaN() since we already made the parameter a number